include/net/xdp.h

```diff
@@ -63,13 +63,30 @@ struct xdp_rxq_info {
 	struct xdp_mem_info mem;
 } ____cacheline_aligned; /* perf critical, avoid false-sharing */
 
+#ifdef CONFIG_XDP_LUA
+struct xdplua_create_work {
+	char				lua_script[8192];
+	struct lua_State	*L;
+	struct work_struct	work;
+	spinlock_t		lock;
+	bool			init;
+};
+
+DECLARE_PER_CPU(struct xdplua_create_work, luaworks);
+#endif /* CONFIG_XDP_LUA */
+
+
 struct xdp_buff {
 	void *data;
 	void *data_end;
 	void *data_meta;
 	void *data_hard_start;
 	unsigned long handle;
 	struct xdp_rxq_info *rxq;
+#ifdef CONFIG_XDP_LUA
```

Comment:
With the creation of `struct xdplua_create_work` there probably is no reason to modify `struct xdp_buff`

Reply:
agreed..